File: java/playwright/src/test/java/dev/selenium/request/MultipleHandlerTest.java
package dev.selenium.request;

import com.microsoft.playwright.Route;
import dev.selenium.TestBase;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Map;

public class MultipleHandlerTest extends TestBase {

  @Test
  public void modifyMultiple() {
    page.route(
        "https://httpbin.org/**",
        route -> {
          Map<String, String> headers = route.request().headers();
          headers.put("X-Test", "true");
          route.resume(new Route.ResumeOptions().setHeaders(headers));
        });

    page.route(
        "https://httpbin.org/**",
        route -> {
          Map<String, String> headers = route.request().headers();
          headers.remove("upgrade-insecure-requests");
          route.fallback(new Route.FallbackOptions().setHeaders(headers));
        });

    page.navigate("https://httpbin.org/headers");

    String bodyText = page.innerText("body").replaceAll("\\s+", "");
    Assertions.assertFalse(bodyText.contains("Upgrade-Insecure-Requests"));
    Assertions.assertTrue(bodyText.contains("\"X-Test\":\"true\""));
  }
}

File: dotnet/Playwright/Request/MultipleHandlerTest.cs
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright.Request;

[TestClass]
public class MultipleHandlerTest : TestBase
{
    [TestMethod]
    public async Task ModifyMultiple()
    {
        await Page!.RouteAsync("https://httpbin.org/**", async route =>
        {
            var headers = new Dictionary<string, string>(await route.Request.AllHeadersAsync());
            headers["X-Test"] = "true";
            await route.ContinueAsync(new RouteContinueOptions { Headers = headers });
        });

        await Page.RouteAsync("https://httpbin.org/**", async route =>
        {
            var headers = new Dictionary<string, string>(await route.Request.AllHeadersAsync());
            headers.Remove("upgrade-insecure-requests");
            await route.FallbackAsync(new RouteFallbackOptions { Headers = headers });
        });

        await Page.GotoAsync("https://httpbin.org/headers");

        var bodyText = await Page.InnerTextAsync("body");
        Assert.IsFalse(bodyText.Contains("Upgrade-Insecure-Requests"));
        Assert.IsTrue(bodyText.Contains("\"X-Test\": \"true\""));
    }
}